contrib/pkg/createcluster/aws.go

```diff
@@ -40,36 +43,58 @@ func (p *awsCloudProvider) generateCredentialsSecret(o *Options) (*corev1.Secret
 	}, nil
 }
 
-func (p *awsCloudProvider) addPlatformDetails(o *Options, cd *hivev1.ClusterDeployment) error {
+func (p *awsCloudProvider) addPlatformDetails(o *Options, cd *hivev1.ClusterDeployment,
+	installConfig *installertypes.InstallConfig, controlPlanePool,
+	computePool *installertypes.MachinePool) error {
+
+	// Inject platform details into ClusterDeployment:
 	cd.Spec.Platform = hivev1.Platform{
 		AWS: &hivev1aws.Platform{
 			Region: "us-east-1",
 		},
 	}
+
 	cd.Spec.PlatformSecrets = hivev1.PlatformSecrets{
 		AWS: &hivev1aws.PlatformSecrets{
 			Credentials: corev1.LocalObjectReference{
 				Name: p.credsSecretName(o),
 			},
 		},
 	}
+
+	cd.Spec.Compute[0].Platform = hivev1.MachinePoolPlatform{
+		AWS: &hivev1aws.MachinePoolPlatform{
+			InstanceType: "m4.xlarge",
+			EC2RootVolume: hivev1aws.EC2RootVolume{
+				IOPS: 100,
+				Size: 22,
+				Type: "gp2",
+			},
+		},
+	}
+
+	// Inject platform details into InstallConfig:
+	installConfig.Platform = installertypes.Platform{
+		AWS: &awsinstallertypes.Platform{
+			Region: "us-east-1",
+		},
+	}
+
 	// Used for both control plane and workers.
-	mpp := &hivev1aws.MachinePoolPlatform{
-		InstanceType: "m4.large",
-		EC2RootVolume: hivev1aws.EC2RootVolume{
+	mpp := &awsinstallertypes.MachinePool{
+		InstanceType: "m4.xlarge",
+		EC2RootVolume: awsinstallertypes.EC2RootVolume{
 			IOPS: 100,
 			Size: 22,
 			Type: "gp2",
 		},
 	}
 
-	cd.Spec.ControlPlane.Platform = hivev1.MachinePoolPlatform{
+	controlPlanePool.Platform = installertypes.MachinePoolPlatform{
```

Comment:
Can this be the following? Or are you trying to ensure also that no other platforms are set?
```
controlPlanePool.Platform.AWS = mpp
computePool.Platform.AWS = mpp
```